### PR TITLE
added ios support

### PR DESCRIPTION
**Pull request by sandeepsj**
Added ios native implementations for canOpenApp, openApp and getResourceByName. Now the webview can call window.HyperWebViewBridge.postMessage and access native functions to use upi apps.
Screen recording: https://drive.google.com/file/d/1jKKuPeD4rgndW3m-jZgxHbsjvV86Blkd/view?usp=sharing

### CODE REVIEW

File: ios/hyper_webview_flutter.podspec [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Pod::Spec.new do |s|
+  s.name             = 'hyper_webview_flutter'
+  s.version          = '0.0.1'
+  s.summary          = 'A new Flutter plugin project.'
+  s.description      = <<-DESC
+A new Flutter plugin project.
+                       DESC
+  s.homepage         = 'http://example.com'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Your Company' => 'email@example.com' }
```
Comment (Subhash703) [inline ios/hyper_webview_flutter.podspec]: @sandeepsj  Please write proper podspec file. You can take reference from **hypersdkflutter**

File: ios/Classes/HyperWebviewFlutterPlugin.m [hunk 2]
```diff
@@ -0,0 +1,106 @@
+#import "HyperWebviewFlutterPlugin.h"
```
Comment (Subhash703) [inline ios/Classes/HyperWebviewFlutterPlugin.m]: @sandeepsj  Can you add video for proof of testing in the PR description?

File: ios/Classes/HyperWebviewFlutterPlugin.m [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+@end
+
+const int OPENAPPS_REQUEST_CODE = 19;
+const int GET_RESOURCE_NAME = 453;
+const int CAN_OPEN_APP_CODE = 789;
+@implementation HyperWebviewFlutterPlugin
+
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
```
Comment (Subhash703) [inline ios/Classes/HyperWebviewFlutterPlugin.m]: @sandeepsj  Why did we decide to use ObjectiveC here? Why not swift?
We are thinking of moving to Swift everywhere. Let's use swift in this new repo

File: ios/Classes/HyperWebviewFlutterPlugin.m [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    HyperWebviewFlutterPlugin* instance = [[HyperWebviewFlutterPlugin alloc] init];
+    instance.dartChannel=[FlutterMethodChannel methodChannelWithName:@"DartChannel" binaryMessenger:[registrar messenger]];
+    FlutterMethodChannel* channel = [FlutterMethodChannel methodChannelWithName:@"NativeChannel" binaryMessenger:[registrar messenger]];
+    instance.nativeChannel=channel;
+    [registrar addMethodCallDelegate:instance channel:channel];
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if ([@"openApp" isEqualToString:call.method]) {
+      [self openApp:[call.arguments objectAtIndex:1]];
```
Comment (Subhash703) [inline ios/Classes/HyperWebviewFlutterPlugin.m]: Let's use firstObject only

File: ios/Classes/HyperWebviewFlutterPlugin.m [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ To Check if an app can be opened by current app
+
+ @param payload App URL
+ */
+- (void)canOpenApp:(NSString*)payload{
+    //Note: canOpen will fail even if app is present in the device if app urls are not
+    //added to info.plist under LSApplicationQueriesSchemes as an array of allowed urls.
+    NSURL *appURL = [NSURL URLWithString:[HyperWebviewFlutterPlugin base64DecodedStringFromString:payload]];
+    BOOL status = [[UIApplication sharedApplication] canOpenURL:appURL];
+    NSString* resPayload = [NSString stringWithFormat:@"{\"result\":\"%@\",\"app\":\"%@\"}", status?@"1":@"0", appURL];
```
Comment (Subhash703) [inline ios/Classes/HyperWebviewFlutterPlugin.m]: let's create an disctionary and convert to String

File: ios/hyper_webview_flutter.podspec (hunk 1 above)
Comment (harshgarg18) [reply ios/hyper_webview_flutter.podspec]: @Subhash703 - to take care in next PR if still open

File: ios/Classes/HyperWebviewFlutterPlugin.m (hunk 2 above)
Comment (harshgarg18) [reply ios/Classes/HyperWebviewFlutterPlugin.m]: @Subhash703 - to take care in next PR if still open

File: ios/Classes/HyperWebviewFlutterPlugin.m (hunk 4 above)
Comment (harshgarg18) [reply ios/Classes/HyperWebviewFlutterPlugin.m]: @Subhash703 - to take care in next PR if still open

File: ios/Classes/HyperWebviewFlutterPlugin.m (hunk 5 above)
Comment (harshgarg18) [reply ios/Classes/HyperWebviewFlutterPlugin.m]: @Subhash703 - to take care in next PR if still open

File: ios/Classes/HyperWebviewFlutterPlugin.m (hunk 3 above)
Comment (harshgarg18) [reply ios/Classes/HyperWebviewFlutterPlugin.m]: @Subhash703 - to take care in next PR if still open

**Pull request merged**